lib/main.js

```diff
@@ -73,20 +72,31 @@ export default {
       }
     }))
     this.subscriptions.add(atom.commands.add('atom-text-editor', {
-      'linter-eslint:fix-file': function() {
+      'linter-eslint:fix-file': () => {
         const textEditor = atom.workspace.getActiveTextEditor()
+        const filePath = textEditor.getPath()
+        const fileDir = Path.dirname(filePath)
+
         if (!textEditor || textEditor.isModified()) {
           // Ignore invalid or unsaved text editors
+          atom.notifications.addError('Linter-ESLint: Please save before fixing')
           return;
         }
-        const filePath = textEditor.getPath()
-        const fileDir = Path.dirname(filePath)
-        new BufferedNodeProcess({
-          command: Path.normalize(Path.join(__dirname, '..', 'node_modules', 'eslint', 'bin', 'eslint.js')),
-          args: [filePath, '--fix'],
-          options: {
-            cwd: fileDir
-          }
+
+        if (this.worker === null) {
+          return;
```

Comment:
Personally I like having them in there, but can live without them. Just as long as the files are consistent :stuck_out_tongue:.